### ISSUE TITLE
Replace Databricks CLI subprocess calls with Databricks SDK for Python?

### ISSUE THREAD

**Issue by astrojuanlu** [JenspederM/kedro-databricks#8]
https://github.com/databricks/databricks-sdk-py/

The Databricks SDK for Python is beta but at least

- it can be installed with pip (unlike the newest Databricks CLI)
- and users would probably get better error messages when things go wrong (instead of cryptic "subprocess exited with status 1" kind of thing)
- and maybe easier to maintain even?

Wdyt @JenspederM ?

**Comment by JenspederM** [comment]
AFAIK, the SDK doesn't support asset bundles. The bundle architecture is more than just deployment, it's also validation and environment separation, all things that would have to be implemented from scratch with the sdk. 

Even validation would have to be implemented separately, as they chose to use dataclasses instead of a proper validation tool such as pydantic. 

I can give it another look, but I feel as if I went pretty deep the first time around 😊

**Comment by astrojuanlu** [comment]
Indeed https://github.com/databricks/databricks-sdk-py/issues/628

Thanks for the quick response! Maybe we can wait a bit longer and see if the issue above is addressed

**Comment by astrojuanlu** [comment]
#75 introduces `databricks-sdk` as a dependency already

**Comment by JenspederM** [comment]
@astrojuanlu I know, but changing the other parts would make it incompatible with asset bundles. And I'm still not convinced that's a smart choice.

As it is now, there is very little lock in as a user can choose to go fully for asset bundles at any time. 

Instead of rewriting the whole system, I'd rather work to improve error handles by forwarding information from the Databricks cli to a larger extend.

**Comment by astrojuanlu** [comment]
Yes yes, it was a cross reference without any attached meaning. The upstream lack of support for DAB is still unaddressed 👍🏼
